ui/src/components/MessageSnackBar.vue

```diff
@@ -0,0 +1,96 @@
+<template>
+  <!-- display a message or error to the end user
+    usage: <MessageSnackBar v-bind:bus="bus"></MessageSnackBar>
+      where bus is a vue instance.
+    do bus.$emit('message', 'message to be displayed')
+    can replace 'message' with 'error' or 'notification' for different style of messages
+    for reference of the event bus model, see https://code.luasoftware.com/tutorials/vuejs/parent-call-child-component-method/
+    -->
+  <div>
+    <!-- TODO: add multiple messages support? 
+      Currently new messages will override the old ones, and count down will not restart <2020-06-04, David Deng> -->
```

Comment:
Good question. My initial guess is that this won't happen to often, so it may not be worth a ton of extra effort right now.